Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/OfferingsList.swift

```diff
@@ -24,62 +24,94 @@ struct OfferingsList: View {
     fileprivate struct PresentedPaywall: Hashable {
         var offering: Offering
         var mode: PaywallViewMode
+        var responseOfferingID: String
     }
 
     @State
-    private var offeringsPaywalls: Result<[OfferingPaywall], NSError>?
+    private var offeringsPaywalls: Result<[OfferingPaywall], NSError>? {
+        didSet {
+            Task { @MainActor in
+                refreshPresentedPaywall()
+            }
+        }
+    }
 
     @State
     private var presentedPaywall: PresentedPaywall?
     
     @State
     private var displayPaywall: Bool = false
-    
-    private let client = HTTPClient.shared
-    
+
     let app: DeveloperResponse.App
 
+    @MainActor
+    private func updateOfferingsAndPaywalls() async {
+        do {
+            let appCopy = app
```

Comment:
We need to copy `self.app` so that sending it into the following static functions doesn't capture `self`, which is a violation of swift concurrency rules since this struct isn't sendable.